leader-election-impl/src/main/java/com/palantir/leader/proxy/AwaitingLeadershipProxy.java

```diff
@@ -84,6 +87,10 @@ private AwaitingLeadershipProxy(Supplier<?> delegateSupplier,
         this.delegateRef = new AtomicReference<Object>();
         this.interfaceClass = interfaceClass;
         this.isClosed = false;
+
+        String logTag = "Proxy #" + numProxies.getAndIncrement() + " - " + interfaceClass.getSimpleName();
+        log = new TaggedLogger(AwaitingLeadershipProxy.class, logTag);
+        leaderLog = new TaggedLogger("leadership", logTag);
```

Comment:
Yeah, I take your point and would like to get rid of TaggedLogger. I'm on the fence about whether it's better to have the dumb TaggedLogger or to do something clever with a custom appender (which I think I'd need since we still want the logs going to the same file). 
I'm _inclined_ to keep it simple and stick with the delegation but I could be persuaded. @gsheasby , what do you reckon?